### ISSUE TITLE
Allow deleting a row from data grid

### ISSUE THREAD

**Issue by prakhargupta1** [mui/mui-toolpad#1039]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

We should allow deleting a row. Ways we could do it:

1. Delete single row: Through an action button as one of the column. Clicking on it should open a confirmation box
2. Delete multiple rows: Through checkboxes as one of the column.
    
Relates to #994

**Comment by bharatkashyap** [comment]
Agreed, @Janpot - we can close these issues now since they relate to the former version of Toolpad

**Comment by prakhargupta1** [comment]
Ok, let's add it docs and then close it.

**Comment by bharatkashyap** [comment]
@prakhargupta1 Can we create a separate issue to document this (and editing rows/adding new rows based on the same pattern)?

**Comment by prakhargupta1** [comment]
About deleting a row, I couldn't make 1. work.
Do I have to create a new query or the one that already brings data to the datagrid?

**Comment by bharatkashyap** [comment]
> About deleting a row, I couldn't make 1. work. Do I have to create a new query or the one that already brings data to the datagrid?

You would have to create a query that deletes a row, right? How would you delete a row with the same query that fetches the rows?

**Comment by prakhargupta1** [comment]
Ok, got it. I guess this is the config:

<img width="1406" alt="Screenshot 2023-05-15 at 9 43 59 AM" src="https://github.com/mui/mui-toolpad/assets/92228082/2a075526-a9d6-4013-8194-24b4a09222b2">

I'll demo deleting a GitHub issue and then add it to docs.

**Comment by bharatkashyap** [comment]
> Ok, got it. I guess this is the config:
> 
> <img alt="Screenshot 2023-05-15 at 9 43 59 AM" width="1406" src="https://user-images.githubusercontent.com/92228082/238241451-2a075526-a9d6-4013-8194-24b4a09222b2.png">
> I'll demo deleting a GitHub issue and then add it to docs.

I think you can also set it to be "Fetch only on manual action"
